File: Library/Web/Controllers/HomeController.cs
using Library.BLL;
using Library.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Web.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        public ActionResult About()
        {
            return View();
        }

        public ActionResult Contact()
        {
            ViewBag.Message = "关于我";

            return View();
        }

        public ActionResult List()
        {
            BookBll bll = new BookBll();
            List<Book> books = bll.GetModelList("");
            ViewBag.books = books;
            return View();
        }

        public ActionResult AddBook()
        {
            return View();
        }


        public ActionResult EditBook(int id)
        {
            BookBll bll = new BookBll();
            if (id > 0)
            {
                Book book = bll.GetModel(id);
                ViewBag.book = book;
            }
            else
            {
                ViewBag.book = null;
            }
            return View();
        }

        public ActionResult DeleteBook(int id)
        {
            BookBll bll = new BookBll();
            if (id > 0)
            {
                Book book = bll.GetModel(id);
                ViewBag.book = book;
            }
            else
            {
                ViewBag.book = null;
            }
            return View();
        }

        public ActionResult BookOperation()
        {
            var bookValues = Request.Form;
            string type = Convert.ToString(bookValues["type"]);
            if(type == "add")
            {
                bool res = this.DealWithAddBook(bookValues);
                ViewBag.message = res ? "添加成功" : "添加失败";
            }

            if (type == "edit")
            {
                bool res = this.DealWithEditBook(bookValues);
                ViewBag.message = res ? "修改成功" : "修改失败";
            }

            if (type == "delete")
            {
                bool res = this.DealWithDeleteBook(bookValues);
                ViewBag.message = res ? "删除成功" : "删除失败";
            }
            return View();
        }

        #region JSON获取
        //public JsonResult AddBook()
        //{
        //    return new JsonResult();
        //}
        #endregion

        #region 私有业务处理逻辑
        private bool DealWithAddBook(dynamic bookValues)
        {
            Book book = new Book()
            {
                name = bookValues["name"],
                author = bookValues["author"],
                publish = bookValues["publish"],
                price = String.IsNullOrEmpty(bookValues["price"]) ? null : Convert.ToDecimal(bookValues["price"]),
                date = DateTime.Now.ToString("yyy-MM-dd HH:mm:ss:f"),
                isbn = bookValues["isbn"],
                cover = bookValues["cover"],
            };

            BookBll bll = new BookBll();
            int res = bll.Add(book);

            return res > 0;
        }


        private bool DealWithEditBook(dynamic bookValues)
        {
            Book book = new Book()
            {
                id = Convert.ToInt32(bookValues["id"]),
                name = bookValues["name"],
                author = bookValues["author"],
                publish = bookValues["publish"],
                price = String.IsNullOrEmpty(bookValues["price"]) ? null : Convert.ToDecimal(bookValues["price"]),
                isbn = bookValues["isbn"],
                cover = bookValues["cover"],
            };

            BookBll bll = new BookBll();
            return bll.Update(book);
        }


        private bool DealWithDeleteBook(dynamic bookValues)
        {
            int id = Convert.ToInt32(bookValues["id"]);

            BookBll bll = new BookBll();
            return bll.Delete(id);
        }


        #endregion

    }
}

File: Subject2/Controller/HomeController.js
import Controller from './Controller';
import BookModel from '../Models/Book';

const booksModel = new BookModel();

class HomeController extends Controller {
    constructor() {
        super()
    }
    async Index(ctx) {
        ctx.body = await ctx.render('Home/Index');
    }
    async List(ctx) {
        const data = await booksModel.getBookList();
        ctx.body = await ctx.render('Home/List', { data: data });
    }
    async AddBook(ctx) {
        ctx.body = await ctx.render('Home/AddBook');
    }
    async EditBook(ctx) {
        // console.log(ctx.params);
        const { id } = ctx.params;
        const data = await booksModel.getBookDetail(id);
        console.log(data);
        ctx.body = await ctx.render('Home/EditBook', { book: data });
    }
    async DeleteBook(ctx) {
        const { id } = ctx.params;
        const data = await booksModel.getBookDetail(id);
        const html = await ctx.render('Home/delbook', { book: data });
        ctx.body = html;
    }
    async OperationResult(ctx) {
        const { type, ...rest } = ctx.request.body;
        console.log(type, rest);
        let res = false;
        if (type === 'add') {
            res = await booksModel.addBook(rest);
        }
        if (type === 'edit') {
            res = await booksModel.updateBook(rest)
        }
        if (type === 'delete') {
            res = await booksModel.deleteBook(rest.id);
        }
        if (res === true) {
            ctx.body = await ctx.render('Home/OperationResult', { message: '操作成功' });
        } else {
            ctx.body = await ctx.render('Home/OperationResult', { message: `操作失败,${JSON.stringify(res)}` });
        }
    }
    async About(ctx) {
        ctx.body = await ctx.render('Home/About');
    }
}

export default HomeController;
